### PR TITLE
Add due date timeline

### PR DESCRIPTION
**Pull request by katjanakosic**
I implemented a DateTimelineView that displays the due dates for release, submission and assessment in the ExerciseListView. It shows the relative date string for each due date. 

Screenshot 2023-02-07 at 12 24 00

### CODE REVIEW

**Review by crustypluto19 — APPROVED** [review]
Looks great!

File: Themis/Views/Exercises/ExerciseListItem.swift [hunk 1]
```diff
@@ -7,26 +7,18 @@
 
 import SwiftUI
 
-struct ExerciseListItem: View {
+ struct ExerciseListItem: View {
```
Comment (huber-florian) [inline Themis/Views/Exercises/ExerciseListItem.swift]: unnecessary space

File: Themis/Views/Exercises/ExerciseListItem.swift [hunk 2]
```diff
@@ -7,26 +7,18 @@
...
-            }
-        }.padding(.trailing)
+            DateTimelineView(dates: dateProperties.map { dateProp in
+                (name: dateProp.name, date: exercise[keyPath: dateProp.dateKeyPath])
+            })
+        }
+        .padding(.trailing)
     }
-}
+ }
```
Comment (huber-florian) [inline Themis/Views/Exercises/ExerciseListItem.swift]: unnecessary space

File: Themis/Views/Exercises/ExerciseView.swift [hunk 3]
```diff
@@ -50,6 +51,14 @@ struct ExerciseView: View {
             )
         }
         .navigationTitle(exercise.title ?? "")
+        .toolbar {
```
Comment (huber-florian) [inline Themis/Views/Exercises/ExerciseView.swift]: Is it really used here?

File: Themis/Views/Exercises/ExerciseListItem.swift (hunk 1 above)
Comment (katjanakosic) [reply Themis/Views/Exercises/ExerciseListItem.swift]: Oh didn't see that. I'll fix it now.

**Review by huber-florian — APPROVED** [review]
Nice, looks great!

**Pull request merged**